datumaro/components/annotation.py

```diff
@@ -421,28 +425,31 @@ def from_instance_masks(
 
         it = iter(masks)
 
+        # Generate an index mask
+        index_mask = None
         instance_map = [0]
-        class_map = [0]
-
-        m, idx, instance_id, class_id = next(it)
-        if not class_id:
-            idx = 0
-        index_mask = make_index_mask(m.image, idx, dtype=index_dtype)
-        instance_map.append(instance_id)
-        class_map.append(class_id)
-
+        class_map = [background_label_id]
         for m, idx, instance_id, class_id in it:
-            if not class_id:
+            if class_id == background_label_id:
```

Comment:
Shouldn't there also be a check for `class_id is None` (as per the docstring)?

Reply:
Added.